.pre-commit-config.yaml

```
@@ -0,0 +1,188 @@
+default_language_version:
+  golang: go1.21
+
+repos:
+  # General file checks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      # Trim trailing whitespace
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+      
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)